### PR TITLE
Prefix Yaml tags with `!bson_`

### PR DESCRIPTION
**Pull request by GromNaN**
After @jmikola's comment: https://github.com/mongodb/mongo-php-builder/pull/37#discussion_r1461971431

### CODE REVIEW

File: generator/src/OperatorTestGenerator.php [hunk 1]
```diff
@@ -138,11 +138,11 @@ private function convertYamlTaggedValues(mixed $object): mixed
...
 
             return match ($object->getTag()) {
-                'regex' => new Regex(...(array) $value),
-                'long' => new Int64($value),
-                'double' => new Decimal128($value),
-                'date' => new UTCDateTime($value),
-                'binary' => new Binary(base64_decode($value)),
+                'bson_regex' => new Regex(...(array) $value),
+                'bson_long' => new Int64($value),
+                'bson_double' => new Decimal128($value),
```
Comment (jmikola) [inline generator/src/OperatorTestGenerator.php]: If we're prefixing these with `bson_`, can we use the actual names for clarity?

That would be `bson_int64` above, `bson_decimal128` here, and `bson_utcdatetime` below.

**Review by jmikola — CHANGES_REQUESTED** [review]

File: generator/src/OperatorTestGenerator.php (hunk 1 above)
Comment (GromNaN) [reply generator/src/OperatorTestGenerator.php]: Are this names shared with other drivers ? The idea is also that this config files could be extracted and used by other languages to create agg builders.

Comment (jmikola) [reply generator/src/OperatorTestGenerator.php]: These names are consistent with the [BSON spec](https://bsonspec.org/spec.html), which pre-dates any driver specs. That's the reason we chose the names when implementing `ext-mongodb` (vs. keeping what we originally had in `ext-mongo`).

Comment (GromNaN) [reply generator/src/OperatorTestGenerator.php]: Ok, renamed.

**Review by jmikola — APPROVED** [review]

**Pull request merged**